### PR TITLE
601 signin page for publish v2

### PR DESCRIPTION
**Pull request by defong**
### Context

signin page for publish

### Changes proposed in this pull request
Removed the basket ball team of booleans.
Added an `intermittent signin page` for publish
Fixed 400+ broken tests * 2 times
Amended azure related settings
Amended readme to reflect this change

### Guidance to review
Pull down the code and go though the 
- 3 mutually exclusive ways to login, 

As well as per the related environment yml vs actual environment expectations:
- defaulted production, 
- defaulted qa,
- defaulted staging,
- defaulted localhost .

The whole user journey has changed. Originally when navigating to any authenticated paths, inclusive of the root path, the user will be redirected to
- `/signin` 

Then they will be further redirected to one of the following:
- `DfE Sign In`
- `Persona`
Or instead 
- the `Magic` page is shown.

Now the user is redirected to `/sign-in` of which the `intermittent Sign In page` will be shown or the `Magic` page is shown, the call to action is a relative button.

The `mutually exclusive` applies from route/endpoint down, these are:
- Non-Production environments
  - `persona`
- Production environments
  - `dfe_signin`
  - `magic_link`

> We managed to introduced 6 various booleans over extended period of time that controls this in very fine grain manner with various implied implicit or explicit expectation. ie you configure those 6 boolean to allow you to log in proper but the same settings fails to log you out, or the settings for productions settings can be easy mistaken and intertwined non-productions settings .

Builds on top of https://github.com/DFE-Digital/publish-teacher-training/pull/1467 whereby @gpeng was the confused survivor of  

> The greater the drift from localhost to mimic a production like settings causes caveats and subtleties and conflicts with the subliming of easy of development, as well as resurrecting the `It works on my machine` heresy.

The PR started as `add gem` then `add sign in page` then :exploding_head: 

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally
- [ ] Product Review

### CODE REVIEW

**Comment by defong** [comment]
[Signin page for publish](https://trello.com/c/xyqQKa2L/601-signin-page-for-publish)

**Comment by defong** [comment]
[Signin page for publish](https://trello.com/c/xyqQKa2L/601-signin-page-for-publish)

**Review by gpeng — CHANGES_REQUESTED** [review]
Just a couple of little comments but this looks awesome. What a headache. We'll need to give this a good run out on QA and try setting the mode from the env. It would be good to have the env variables set up, even though the config is in here. We'll also need to add some documentation for switching mode.

File: config/settings/production.yml [hunk 1]
```diff
@@ -11,3 +11,8 @@ search_ui:
 environment:
   label: "Beta"
   selector_name: "beta"
+
+# # Wield magic
```
[the remark this answers is not in the document]
Comment (defong) [reply config/settings/production.yml:15]: It's __explicit__ by `default` in the same veins as __secure__ by `default`

Changing the `mode` is an operation concerns, this only serves as a comment and last resort.

**Review by gpeng — CHANGES_REQUESTED** [review]
Can we change the references to `magic` and `magic?` to `magic_link`. Or maybe `email_link?`. Logging in by `magic` doesn't seem very helpful 😆 I think this is looking good now though. I still feel we should be explicit in each environment so as it's clear what the default is. It's an important feature.

File: config/settings/production.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (defong) [reply config/settings/production.yml:18]: Also added to file now

**Comment by defong** [comment]
> Can we change the references to `magic` and `magic?` to `magic_link`. Or maybe `email_link?`. Logging in by `magic` doesn't seem very helpful laughing I think this is looking good now though. I still feel we should be explicit in each environment so as it's clear what the default is. It's an important feature.

Fair enough, but we are so so so keeping this method

https://github.com/DFE-Digital/publish-teacher-training/blob/7b856accff17a22ca372a52209d73a21aef77a1f/app/controllers/sessions_controller.rb#L31

**Pull request merged**